### PR TITLE
Add documentation for JSON output endpoint

### PR DESCRIPTION
**Pull request by jondavidjohn**
## Description

We recently added an endpoint to allow users to retrieve a [json formatted execution plan](https://www.terraform.io/docs/internals/json-format.html#format-summary). This documents that new endpoint.

### CODE REVIEW

File: content/source/docs/cloud/api/plans.html.md [hunk 1]
```diff
@@ -77,8 +78,40 @@ curl \
...
 }
 ```
+
+## Retrieve the JSON execution plan
+
+`GET /plans/:id/json-output`
+
+`GET /runs/:id/plan/json-output`
+
+These endpoints generate a temporary authenticated URL to the location of the [JSON formatted execution plan](https://www.terraform.io/docs/internals/json-format.html#format-summary).  When successful, this endpoint responds with a temporarily redirect that should be followed.  If using a client that can follow redirects, you can use this endpoint to save the `.json` file locally without needing to save the temporary URL.
```
Comment (chrisarcand) [inline content/source/docs/cloud/api/plans.html.md:94]:
```suggestion
These endpoints generate a temporary authenticated URL to the location of the [JSON formatted execution plan](https://www.terraform.io/docs/internals/json-format.html#format-summary).  When successful, this endpoint responds with a temporary redirect that should be followed.  If using a client that can follow redirects, you can use this endpoint to save the `.json` file locally without needing to save the temporary URL.
```

File: content/source/docs/cloud/api/plans.html.md [hunk 2]
```diff
@@ -77,8 +78,40 @@ curl \
...
+
+-> **Note:** This endpoint is available for plans using Terraform 0.12 and later.
+
+-> **Note:** This endpoint cannot be accessed with [organization tokens](../users-teams-organizations/api-tokens.html#organization-api-tokens). You must access it with a [user token](../users-teams-organizations/users.html#api-tokens) or [team token](../users-teams-organizations/api-tokens.html#team-api-tokens) that has [**admin** level permission](../users-teams-organizations/permissions.html#admin) to access the workspace.
+
+Status  | Response                  | Reason
+--------|---------------------------|----------
+[204][] | No Content                | Plan JSON supported, but plan has not yet completed.
+[307][] | Temporary Redirect        | Plan JSON found and temporary download URL generated
+[422][] | [JSON API error object][] | Plan does not use a supported version of terraform (< 0.12.X)
```
Comment (chrisarcand) [inline content/source/docs/cloud/api/plans.html.md:106]: "Is '< 0.12.X' the supported or unsupported version, huh?"

"Plan was generated with a earlier version of Terraform than 0.12 (unsupported)" maybe?

Comment (jondavidjohn) [reply content/source/docs/cloud/api/plans.html.md:106]: Maybe the specific version callout here is unnecessary, I also like your phrasing better.

Going with

"Plan was generated with an unsupported version of Terraform"

**Comment by jondavidjohn** [comment]
@nfagerlund Even though this was reviewed once already, I'd still love to get your eyes on it whenever you get the time!

File: content/source/docs/cloud/api/plans.html.md [hunk 3]
```diff
@@ -77,8 +78,40 @@ curl \
...
+
+These endpoints generate a temporary authenticated URL to the location of the [JSON formatted execution plan](https://www.terraform.io/docs/internals/json-format.html#format-summary).  When successful, this endpoint responds with a temporary redirect that should be followed.  If using a client that can follow redirects, you can use this endpoint to save the `.json` file locally without needing to save the temporary URL.
+
+This temporary URL provided by the redirect has a life of **1 minute**, and should not be relied upon beyond the initial request.  If you need repeat access, you should use this endpoint to generate a new URL each time.
+
+-> **Note:** This endpoint is available for plans using Terraform 0.12 and later.
+
+-> **Note:** This endpoint cannot be accessed with [organization tokens](../users-teams-organizations/api-tokens.html#organization-api-tokens). You must access it with a [user token](../users-teams-organizations/users.html#api-tokens) or [team token](../users-teams-organizations/api-tokens.html#team-api-tokens) that has [**admin** level permission](../users-teams-organizations/permissions.html#admin) to access the workspace.
```
Comment (nfagerlund) [inline content/source/docs/cloud/api/plans.html.md:100]:
```suggestion
-> **Note:** This endpoint cannot be accessed with [organization tokens](../users-teams-organizations/api-tokens.html#organization-api-tokens). You must access it with a [user token](../users-teams-organizations/users.html#api-tokens) or [team token](../users-teams-organizations/api-tokens.html#team-api-tokens) that has [**admin** level permission](../users-teams-organizations/permissions.html#admin) for the workspace.
```

File: content/source/docs/cloud/api/plans.html.md (hunk 2 above)
Comment (nfagerlund) [reply content/source/docs/cloud/api/plans.html.md:106]: I'm in favor of "Plan was generated with an unsupported version of Terraform".

**Pull request merged**